drivers/adodb-ldap.inc.php

```diff
@@ -150,48 +193,90 @@ function _connect( $host, $username, $password, $ldapbase)
 	);
 */
 
-	function _inject_bind_options( $options ) {
-		foreach( $options as $option ) {
-			ldap_set_option( $this->_connectionID, $option["OPTION_NAME"], $option["OPTION_VALUE"] )
-				or die( "Unable to set server option: " . $option["OPTION_NAME"] );
+	/**
+	 * Converts old style global options into connection parameters
+	 *
+	 * @deprecated 5.23
+	 * @param array $options
+	 * @return void
+	 */
+	private function _inject_bind_options( $options ) {
+		foreach( $options as $option ) 
+		{
+			$this->connectionParameters[$option["OPTION_NAME"]] = $option["OPTION_VALUE"];
+			//ldap_set_option( $this->_connectionID, $option["OPTION_NAME"], $option["OPTION_VALUE"] );
+			//	or die( "Unable to set server option: " . $option["OPTION_NAME"] );
 		}
 	}
 
-	function _query($sql,$inputarr=false)
-	{
-		$rs = @ldap_search( $this->_connectionID, $this->database, $sql );
-		$this->_errorMsg = ($rs) ? '' : 'Search error on '.$sql.': '.ldap_error($this->_connectionID);
-		return $rs;
-	}
 
-	function ErrorMsg()
+	/**
+	 * Execute a query against the AD Database.
+	 *
+	 * @param string|array $ldapQuery   Query to execute.
+	 * @param array        $inputarr   [Ignored] An optional array of parameters.
+	 *
+	 * @return ldap_query|bool
+	*/
+	function _query($ldapQuery,$inputarr=false)
 	{
-		return $this->_errorMsg;
+		
+		$rs = ldap_search( $this->_connectionID, $this->database, $ldapQuery );
+		
+		$this->_errorMsg = ($rs) ? '' : 'Search error on '.$ldapQuery.': '.ldap_error($this->_connectionID);
+		
+		return $rs;
 	}
 
-	function ErrorNo()
+	
+	/**
+	 * Returns the last error number from previous AD operation.
+	 *
+	 * @return int The last error number.
+	 */
+	function errorNo()
 	{
 		return @ldap_errno($this->_connectionID);
 	}
 
-	/* closes the LDAP connection */
+	/**
+	 * closes the LDAP connection 
+	 *
+	 * @return void
+	 */ 
 	function _close()
 	{
-		@ldap_close( $this->_connectionID );
+		if (is_resource($this->_connectionID))
+			@ldap_close( $this->_connectionID );
+
 		$this->_connectionID = false;
 	}
 
-	function SelectDB($db) {
-		$this->database = $db;
+	/**
+	 * Switches the baseDN to a new value
+	 *
+	 * @param string $baseDN
+	 * @return void
```

Comment:
Return type does not match, should be bool / true, or remove the `return true;` statement below.

Reply:
Fixed return definition, improved docblock